WooCommerce/WooCommerceUITests/Tests/ProductsTests.swift

```diff
@@ -0,0 +1,29 @@
+import UITestsFoundation
+import XCTest
+
+final class ProductsTests: XCTestCase {
+
+    override func setUpWithError() throws {
+        continueAfterFailure = false
+
+        let app = XCUIApplication()
+        app.launchArguments = ["logout-at-launch", "disable-animations", "mocked-wpcom-api", "-ui_testing"]
+        app.launch()
+        try LoginFlow.logInWithWPcom()
+    }
+
+    func testProductsScreenLoad() throws {
+        try TabNavComponent().gotoProductsScreen()
+        XCTAssert(try ProductsScreen().isLoaded)
+
+        let products = try GetMocks.readProductsData()
+        let firstProductName = products.response.jsonBody.data[0].name
+        try ProductsScreen().verifyProductNameOnProductsScreen(name: firstProductName)
+
+        try ProductsScreen().selectProduct(atIndex: 0)
```

Comment:
I'd suggest adding another method which selects a product by name. It can be also named `selectProduct` but have another parameter, something like `productName`, Swift allows that. Why I thought of that is because currently the test relies on the assumption that the order of products in the list is the same are the order of mocks data. While we're more or less sure that this will always be the case, relying on names instead of indices will remove any hesitations on that matter.

Reply:
> The default sort order is by name A-Z which is how the mocked data is sorted at the moment so using index value should be fine. If we use name instead of index, we wouldn't be able to confirm that it's sorted correctly.

This is a good point. How I see it, is that our objective here is to test the app UI, and be abstracted from the API, which we achieve by using mocks. If the app is using A-Z sorting, it should be so, no matter how the mocked data itself is sorted (in fact, the sorting of data in mocks in not something we test or rely on, and honestly I'm not sure if A-Z sorting in mocks is provided by Charles, or it comes this way from API).

If we talk about a sorting test, it should make sure that the ordering in UI is really A-Z.

The issue I see with the current approach in test, is that it's not checking that the selected product (line 23) is really the product we searched for (line 21). It instead checks that there is an element with `firstProductName` label somewhere on the screen, and then opens the very first product. But in fact, the test doesn't know which product it opens. If we imagine that the app is totally broken, and randomly mixes the products in the list, and also shows a random product on a single product screen (no matter which product was tapped), there is a (very low) chance that such test will still pass. 🤔

Reply:
Okay I'm starting to see your point - I tested out the sort filters and it has no relation to the API response sent by the mocked data. In the current test, it is unlikely to fail as we have control over the mocked data and it is sorted alphabetically and default sort is alphabetically ascending so if the test fails it will be a legit failure. 

That said and now that I think more about it, two things can happen - mock data can change and the default sort order can change as well. In which case you are right, the first item on the list may no longer be the first item on the mock file and the test may no longer be correct. I'll create a new method to select based on product name and update this. Will keep this one for filters test where we'd want to check it is sorted as expected.